app/models/user.rb

```diff
@@ -47,7 +58,8 @@ class User < ActiveRecord::Base
   # This will also let us return a human error message.
   #
   def self.authenticate(login, password)
-    user = find_by_login(login.downcase) || find_by_email(login.downcase) # need to get the salt
+    user = find_by_login(login.downcase) ||
+           User.maycamp.find_by_email(login.downcase) # need to get the salt
```

Comment:
This new `maycamp` scope is nice. I am wondering if it is a good idea to have a spec for it...